src/main/java/io/cdap/plugin/gcp/firestore/sink/FirestoreRecordWriter.java

```diff
@@ -58,16 +59,36 @@ public class FirestoreRecordWriter extends RecordWriter<NullWritable, Map<String
    */
   public FirestoreRecordWriter(TaskAttemptContext taskAttemptContext) {
     Configuration config = taskAttemptContext.getConfiguration();
+
     String projectId = config.get(FirestoreConfig.NAME_PROJECT);
+    String databaseId = config.get(FirestoreConfig.NAME_DATABASE);
+    Boolean isServiceAccountFilePath = false;
+    String serviceAccountJson = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_JSON);
     String serviceAccountFilePath = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
     String collection = Strings.nullToEmpty(config.get(FirestoreConstants.PROPERTY_COLLECTION)).trim();
+    // Get Service Account whether JSON or FilePath
+    String serviceAccountType = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_TYPE);
+
+    LOG.debug("Initialize RecordReader(projectId={}, databaseId={}, collection={}, serviceAccountType={}, isServiceAccountFilePath={}, serviceFilePath={}, " +
+    "serviceAccountJson={}", projectId, databaseId, collection, serviceAccountType, isServiceAccountFilePath, serviceAccountFilePath, serviceAccountJson);
+    String serviceAccount = "";
+    if(serviceAccountType.equalsIgnoreCase("FilePath")) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
+      isServiceAccountFilePath = true;
+    } else if(serviceAccountType.equalsIgnoreCase("json")) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_JSON);
+      isServiceAccountFilePath = false;
+    } else {
+      throw new FirestoreInitializationException("Service account type can only be either a File Path or JSON.");
+    }
```

Comment:
In the `FirestoreConfig` class, add these methods which you can use directly here and in other places:
```
@Nullable
  public Boolean isServiceAccountJson() {
    String serviceAccountType = getServiceAccountType();
    return Strings.isNullOrEmpty(serviceAccountType) ? null : serviceAccountType.equals(SERVICE_ACCOUNT_JSON);
  }
  ```
  ```
  @Nullable
  public Boolean isServiceAccountFilePath() {
    String serviceAccountType = getServiceAccountType();
    return Strings.isNullOrEmpty(serviceAccountType) ? null : serviceAccountType.equals(SERVICE_ACCOUNT_FILE_PATH);
  }
  ```
  
  ```
  @Nullable
  public String getServiceAccount() {
    Boolean serviceAccountJson = isServiceAccountJson();
    if (serviceAccountJson == null) {
      return null;
    }
    return serviceAccountJson ? getServiceAccountJson() : getServiceAccountFilePath();
  }
  ```

Reply:
@itsankit-google - I had to revert this change to the original because after using the `FirestoreConfig` methods in let's say `FirestoreRecordReader`, the values are always `null`. The only way to get a value is via the above i.e. `config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH)`. This causes the pipelines to never see the credentials and always fail.

Reply:
On a quick look, I think I confused Hadoop config class with plugin config class. Will review the PR by EOD.

Reply:
Yep, this uses the Hadoop plugin class to fetch the configuration and properties. Thanks!!